packages/connect/src/device/Device.ts

```diff
@@ -595,6 +606,47 @@ export class Device extends TypedEmitter<DeviceEvents> {
         }
     }
 
+    async checkFirmwareHash(): Promise<FirmwareHashCheckResult> {
+        const createFailResult = (error: FirmwareHashCheckError) => ({ success: false, error });
+
+        // optional setting, see types/settings.ts
+        const baseUrl = DataManager.getSettings('binFilesBaseUrl');
```

Comment:
dependency on external module. maybe we want to get rid of those slowly? not sure.

Reply:
What do you mean by external? `DataManager` is part of `connect` :thinking: 
And I'm not yet very familiar with `connect`, so I can't really say what's desired :smile:

Reply:
we are trying to decouple logic and make it flexible by passing data in params instead of relying on global scope. But in this module (Device.ts) DataMangare is already used this way, so its ok.